src/stdlib_selection.fypp

```diff
@@ -0,0 +1,270 @@
+#:include "common.fypp"
+! Specify kinds/types for the input array in select and arg_select
+#:set ARRAY_KINDS_TYPES = INT_KINDS_TYPES + REAL_KINDS_TYPES
+! The index arrays are of all INT_KINDS_TYPES
+
+module stdlib_selection
+!
+! This code was modified from the "Coretran" implementation "quickSelect" by
+! Leon Foks, https://github.com/leonfoks/coretran/tree/master/src/sorting
+!
+! Leon gave permission for this to be released under stdlib's MIT licence.
+! (https://github.com/fortran-lang/stdlib/pull/500#commitcomment-57418593)
+!
+
+use stdlib_kinds, only: int8, int16, int32, int64, sp, dp, qp
+
+implicit none
+
+private
+
+public select, arg_select
+
+interface select
+  #:for arraykind, arraytype in ARRAY_KINDS_TYPES
+    #:for intkind, inttype in INT_KINDS_TYPES
+      #:set name = rname("select", 1, arraytype, arraykind, intkind)
+      module procedure ${name}$
+    #:endfor
+  #:endfor
+end interface
+
+interface arg_select
+  #:for arraykind, arraytype in ARRAY_KINDS_TYPES
+    #:for intkind, inttype in INT_KINDS_TYPES
+      #:set name = rname("arg_select", 1, arraytype, arraykind, intkind)
+      module procedure ${name}$
+    #:endfor
+  #:endfor
+end interface
```

Comment:
FORD meta-data (`version: `) and links (`Specifications`) are missing.

Reply:
Applied in recent commit. I was not certain about the Specification's link-structure, and after looking at a few examples, decided to change the sub-heading structure in the specs (so it now matches that used in stdlib_quadrature, and the links are also analogous).